converse.js

```diff
@@ -634,11 +635,35 @@
             },this), 200));
         };
 
+        this.ping = function (jid, timeout, success, error){
+            if (typeof jid === 'undefined'     || jid == null)     { jid = this.bare_jid; }
+                if (typeof timeout === 'undefined' || timeout == null) { timeout = 45; }
+                if (typeof success === 'undefined' ) { success = null; }
+                if (typeof error === 'undefined' ) { error = null; }
+                if (this.connection) this.connection.ping.ping( jid, success, error, timeout );
+        };
+
+        this.pong = function (ping){
+		    converse.connection.ping.pong(ping);
+	        return true;
+    	};
+
+	    this.registerPongHandler = function (){
+			converse.connection.ping.addPingHandler( this.pong );
+    	};
+
+	    this.registerPingHandler = function (){
+            if (this.ping_interval>0){
+		        window.setInterval( function() { converse.ping(); }, this.ping_interval*1000);
+                }
+    	};
```

Comment:
If we change every converse.connection.send and converse.connection.sendIQ to internal functions converse.send and converse.sendIQ, it should be possible to keep a 'lastMessageTimeStamp' to compare to clock every X seconds.

There is also http://strophe.im/strophejs/doc/1.2.0/files/strophe-js.html#Strophe.Connection.addTimedHandler instead of window.setInterval.